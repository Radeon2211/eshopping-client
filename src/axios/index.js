import axios from 'axios';
import adapter from 'axios/lib/adapters/http';

const serverURL = 'https://radeon2211-eshopping.herokuapp.com';
const localURL = 'http://192.168.1.109:4000';

export const baseURL = serverURL;

const instance = axios.create({
  baseURL,
  withCredentials: true,
  adapter,
});

const getCsrfToken = async () => {
  const { data } = await instance.get('/csrf-token');
  instance.defaults.headers.post['X-CSRF-Token'] = data.csrfToken;
};
getCsrfToken();

export default instance;
