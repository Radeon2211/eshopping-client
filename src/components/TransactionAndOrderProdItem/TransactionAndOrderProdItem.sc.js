import styled from 'styled-components';
import FlexWrapper from '../UI/FlexWrapper';

export const Wrapper = styled(FlexWrapper)`
  padding-top: ${({ theme }) => theme.spacings.level3};

  &:last-child {
    padding-bottom: 0;
  }
`;

export const NameAndPrice = styled.div`
  align-items: center;
  display: flex;
  flex: 1;

  & > *:not(:last-child) {
    margin-right: ${({ theme }) => theme.spacings.level1};
  }

  & .name {
    flex: 2;
    font-size: ${({ theme }) => theme.fontSizes.level3};

    & > a {
      transition: color ${({ theme }) => theme.durations.level1}s;

      &:hover {
        color: ${({ theme }) => theme.colors.green};
      }
    }
  }

  & .overall-price {
    font-size: ${({ theme }) => theme.fontSizes.level4};
  }

  & .price-per-piece {
    font-size: ${({ theme }) => theme.fontSizes.level2};
  }

  @media only screen and (max-width: 56.25em) {
    align-items: start;
    flex-direction: column;
    justify-content: center;

    & > *:not(:last-child) {
      margin-right: 0;
      margin-bottom: ${({ theme }) => theme.spacings.level1};
    }

    & .name {
      flex: initial;
    }
  }
`;

export const Price = styled(FlexWrapper)`
  @media only screen and (max-width: 56.25em) {
    flex: initial;
  }
`;
