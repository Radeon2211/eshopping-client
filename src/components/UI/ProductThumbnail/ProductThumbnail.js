import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import FlexWrapper from '../FlexWrapper';
import { validateURL } from '../../../shared/utility/utility';
import noPhoto from '../../../images/no-photo.png';

const SC = {};
SC.Wrapper = styled(FlexWrapper)`
  max-height: ${({ height }) => height}rem;
  min-width: ${({ width }) => width}rem;
  width: ${({ width }) => width}rem;

  & img {
    max-height: ${({ height }) => height}rem;
    max-width: ${({ width }) => width}rem;
    object-fit: cover;
  }
`;

const ProductThumbnail = React.memo((props) => {
  const { photo, alt, productId, width, height, orderId } = props;

  const photoURL = orderId
    ? `${process.env.REACT_APP_API_URL}/orders/${orderId}/${productId}/photo`
    : `${process.env.REACT_APP_API_URL}/products/${productId}/photo`;
  const validPhotoURL = validateURL(photoURL) ? photoURL : noPhoto;

  return (
    <SC.Wrapper
      align="center"
      justify="center"
      width={width}
      height={height}
      data-testid="ProductThumbnail"
    >
      <img src={photo ? validPhotoURL : noPhoto} alt={alt} data-testid="ProductThumbnail-img" />
    </SC.Wrapper>
  );
});

ProductThumbnail.defaultProps = {
  orderId: '',
};

ProductThumbnail.propTypes = {
  photo: PropTypes.bool.isRequired,
  alt: PropTypes.string.isRequired,
  productId: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired,
  orderId: PropTypes.string,
};

export default ProductThumbnail;
