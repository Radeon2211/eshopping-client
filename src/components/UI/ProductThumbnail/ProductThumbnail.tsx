import React from 'react';
import styled from 'styled-components';
import { validateURL } from '../../../shared/utility/utility';
import noPhoto from '../../../images/no-photo.png';

interface ProductThumbnailProps {
  isPhoto: boolean;
  alt: string;
  productId: string;
  width: string;
  height: string;
  orderId?: string;
}

const Wrapper = styled.div<{ $width: string; $height: string; 'data-testid': string }>`
  display: flex;
  align-items: center;
  min-height: 100%;
  max-height: ${({ $height }) => $height}rem;
  min-width: ${({ $width }) => $width}rem;
  width: ${({ $width }) => $width}rem;

  & > img {
    max-height: ${({ $height }) => $height}rem;
    max-width: ${({ $width }) => $width}rem;
    object-fit: cover;
  }
`;

const ProductThumbnail = React.memo(
  ({ isPhoto, alt, productId, width, height, orderId }: ProductThumbnailProps) => {
    const photoURL = orderId
      ? `${process.env.REACT_APP_API_URL}/orders/${orderId}/${productId}/photo`
      : `${process.env.REACT_APP_API_URL}/products/${productId}/photo`;
    const validPhotoURL = validateURL(photoURL) ? photoURL : noPhoto;

    return (
      <Wrapper $width={width} $height={height} data-testid="ProductThumbnail">
        <img src={isPhoto ? validPhotoURL : noPhoto} alt={alt} data-testid="ProductThumbnail-img" />
      </Wrapper>
    );
  },
);

export default ProductThumbnail;

ProductThumbnail.displayName = 'ProductThumbnail';
