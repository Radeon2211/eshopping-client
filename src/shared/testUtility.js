// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const historyPageNum = (pageNumber) => ({
  listen: jest.fn(),
  location: { search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push: jest.fn(),
});

export const propsPagination = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  maxQuantityPerPage: 2,
});
