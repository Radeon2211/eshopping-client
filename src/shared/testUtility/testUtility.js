// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';
import { v4 as uuidv4 } from 'uuid';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../../store/reducers/rootReducer';
import { productConditions } from '../constants';

export const defaultDeliveryAddress = {
  firstName: 'firstName',
  lastName: 'lastName',
  street: 'street',
  zipCode: 'zipCode',
  city: 'city',
  country: 'country',
  phone: 'phone',
};

export const defaultUserProfile = {
  ...defaultDeliveryAddress,
  _id: 'userId',
  username: 'username',
  email: 'email@domain.com',
  status: 'active',
  contacts: {
    email: true,
    phone: true,
  },
  createdAt: '2021-02-10T19:10:38.872Z',
};

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const createHistoryPageNumber = (pageNumber = 1, push = jest.fn()) => ({
  listen: jest.fn(),
  location: { pathname: '/products', search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push,
});

export const createPaginationProps = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  quantityPerPage: 2,
});

export const createProductItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
  quantitySold = 0,
  buyerQuantity = 0,
  description = '',
  condition = productConditions.NEW,
) => ({
  _id: productId || uuidv4(),
  name,
  price,
  quantity,
  quantitySold,
  buyerQuantity,
  description,
  condition,
  photo,
  seller: {
    username: sellerUsername,
  },
  __v: 0,
  createdAt: '2021-02-10T19:10:38.872Z',
  updatedAt: '2021-02-10T19:10:38.872Z',
});

export const createCartItem = (
  sellerUsername = 'username',
  quantity = 1,
  productId,
  price = 2,
  name = 'product name',
  productQuantity = 2,
  photo = false,
) => {
  const product = createProductItem(
    productId || uuidv4(),
    sellerUsername,
    productQuantity,
    price,
    name,
    photo,
  );

  return {
    _id: uuidv4(),
    quantity,
    product,
  };
};

export const createTransactionAndOrderProdItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
) => ({
  _id: productId || uuidv4(),
  name,
  photo,
  price,
  quantity,
  seller: {
    username: sellerUsername,
  },
});

export const createOrder = (
  products = [createTransactionAndOrderProdItem()],
  orderId = uuidv4(),
  sellerUsername,
  buyerUsername,
  overallPrice,
  createdAt,
  sellerEmail,
  sellerPhone,
) => {
  const seller = sellerUsername
    ? {
        username: sellerUsername,
        email: sellerEmail,
        phone: sellerPhone,
      }
    : null;

  const buyer = buyerUsername
    ? {
        username: buyerUsername,
      }
    : null;

  return {
    _id: orderId,
    seller,
    buyer,
    products,
    overallPrice,
    deliveryAddress: defaultDeliveryAddress,
    createdAt,
    updatedAt: createdAt,
    __v: 0,
  };
};

export const testStore = (initialAuth = {}, initialProduct = {}, initialUI = {}) => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
  const defaultState = createStoreWithMiddleware(rootReducer).getState();

  const finalState = {
    auth: {
      ...defaultState.auth,
      ...initialAuth,
    },
    product: {
      ...defaultState.product,
      ...initialProduct,
    },
    ui: {
      ...defaultState.ui,
      ...initialUI,
    },
  };
  const finalStore = createStoreWithMiddleware(rootReducer, finalState);

  return {
    store: finalStore,
    initialState: finalStore.getState(),
  };
};

export const checkReqMethodAndURL = (moxiosInstance, method, url, reqNumber = 0) => {
  if (reqNumber) {
    const requests = moxiosInstance.requests.__items;
    const prevRequest = requests[reqNumber - 1];
    return method === prevRequest.config.method && url === prevRequest.config.url;
  }
  const lastRequest = moxiosInstance.requests.mostRecent();
  return method === lastRequest.config.method && url === lastRequest.config.url;
};

export const createExpectedState = (
  initialState,
  expectedAuth = {},
  expectedProduct = {},
  expectedUI = {},
) => {
  const newState = {
    auth: {
      ...initialState.auth,
      ...expectedAuth,
    },
    product: {
      ...initialState.product,
      ...expectedProduct,
    },
    ui: {
      ...initialState.ui,
      ...expectedUI,
    },
  };

  return newState;
};

export const setUpStoreWithDefaultProfile = (auth = {}, product = {}, ui = {}) => {
  return testStore(
    {
      profile: defaultUserProfile,
      deliveryAddress: defaultDeliveryAddress,
      cart: [],
      ...auth,
    },
    product,
    ui,
  );
};
