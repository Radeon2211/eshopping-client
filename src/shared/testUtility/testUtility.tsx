/* eslint-disable import/no-extraneous-dependencies */
import { ReactElement } from 'react';
import { Router } from 'react-router-dom';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { MockStoreEnhanced } from 'redux-mock-store';
import { fireEvent, waitFor, render } from '@testing-library/react';
import { v4 as uuidv4 } from 'uuid';
import moxios from 'moxios';
import rootReducer from '../../store/reducers/rootReducer';
import theme from '../../styled/theme';
import {
  CartItem,
  DeliveryAddress,
  Order,
  OrderProductItem,
  Product,
  Profile,
  ProfileStatus,
} from '../types/types';
import { ProductCondition } from '../types/enums';

export const defaultDeliveryAddress: DeliveryAddress = {
  firstName: 'firstName',
  lastName: 'lastName',
  street: 'street',
  zipCode: 'zipCode',
  city: 'city',
  country: 'country',
  phone: 'phone',
};

export const defaultUserProfile: Profile = {
  ...defaultDeliveryAddress,
  _id: 'userId',
  username: 'username',
  email: 'email@domain.com',
  status: ProfileStatus.ACTIVE,
  contacts: {
    email: true,
    phone: true,
  },
  createdAt: '2021-02-10T19:10:38.872Z',
};

export const createPaginationProps = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  quantityPerPage: 2,
});

export const createProductItem = (props: Partial<Product> = {}): Product => {
  return {
    _id: uuidv4(),
    quantity: 1,
    price: 2,
    name: 'product name',
    photo: false,
    quantitySold: 0,
    buyerQuantity: 0,
    description: '',
    condition: ProductCondition.NEW,
    seller: {
      username: 'username',
    },
    createdAt: '2021-02-10T19:10:38.872Z',
    updatedAt: '2021-02-10T19:10:38.872Z',
    __v: 0,
    ...props,
  };
};

export const createCartItem = (props = {}): CartItem => {
  const finalProps = {
    id: uuidv4(),
    sellerUsername: 'username',
    quantity: 1,
    productId: uuidv4(),
    price: 2,
    name: 'product name',
    productQuantity: 2,
    photo: false,
    ...props,
  };
  const product = createProductItem({
    _id: finalProps.productId,
    quantity: finalProps.productQuantity,
    price: finalProps.price,
    name: finalProps.name,
    photo: finalProps.photo,
    seller: {
      username: finalProps.sellerUsername,
    },
  });

  return {
    _id: finalProps.id,
    quantity: finalProps.quantity,
    product,
  };
};

export const createTransactionAndOrderProdItem = (
  props: Partial<OrderProductItem> = {},
): OrderProductItem => {
  return {
    _id: uuidv4(),
    quantity: 1,
    price: 2,
    name: 'product name',
    photo: false,
    seller: {
      username: 'username',
    },
    ...props,
  };
};

export const createOrder = (props: Partial<Order> = {}): Order => {
  return {
    _id: uuidv4(),
    overallPrice: 100,
    seller: null,
    buyer: null,
    products: [createTransactionAndOrderProdItem()],
    deliveryAddress: defaultDeliveryAddress,
    createdAt: '2021-02-10T19:10:38.872Z',
    updatedAt: '2021-02-10T19:10:38.872Z',
    __v: 0,
    ...props,
  };
};

export const testStore = (initialAuth = {}, initialProduct = {}, initialUI = {}) => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
  const defaultState = createStoreWithMiddleware(rootReducer).getState();

  const finalState = {
    auth: {
      ...defaultState.auth,
      ...initialAuth,
    },
    product: {
      ...(defaultState.product as object),
      ...initialProduct,
    },
    ui: {
      ...(defaultState.ui as object),
      ...initialUI,
    },
  };
  // @ts-expect-error TBF after redux upgrade
  const finalStore = createStoreWithMiddleware(rootReducer, finalState);

  return {
    store: finalStore,
    initialState: finalStore.getState(),
  };
};

type MoxiosRequestsItems = { config: { method: string; url: string } }[];
type MockedMoxiosInstance = {
  requests: {
    __items?: MoxiosRequestsItems;
    mostRecent?: () => { config: { method: string; url: string } };
  };
};
type DefaultMoxiosInstance = Omit<typeof moxios, 'requests'> & {
  requests: typeof moxios.requests & {
    __items?: MoxiosRequestsItems;
  };
};
export const checkReqMethodAndURL = (
  moxiosInstance: MockedMoxiosInstance | DefaultMoxiosInstance,
  method: string,
  url: string,
  reqNumber = 0,
) => {
  if (reqNumber) {
    const requests = moxiosInstance.requests.__items;
    const prevRequest = requests?.[reqNumber - 1];
    return prevRequest && method === prevRequest.config.method && url === prevRequest.config.url;
  }
  if (!moxiosInstance.requests.mostRecent) return false;
  const lastRequest = moxiosInstance.requests.mostRecent();
  return method === lastRequest.config.method && url === lastRequest.config.url;
};

export const createExpectedState = (
  initialState: {
    auth?: object;
    product?: object;
    ui?: object;
  },
  expectedAuth = {},
  expectedProduct = {},
  expectedUI = {},
) => {
  const newState = {
    auth: {
      ...initialState.auth,
      ...expectedAuth,
    },
    product: {
      ...initialState.product,
      ...expectedProduct,
    },
    ui: {
      ...initialState.ui,
      ...expectedUI,
    },
  };

  return newState;
};

export const setUpStoreWithDefaultProfile = (auth = {}, product = {}, ui = {}) => {
  return testStore(
    {
      profile: defaultUserProfile,
      deliveryAddress: defaultDeliveryAddress,
      cart: [],
      ...auth,
    },
    product,
    ui,
  );
};

export const clickAtSubmitButton = async (container: Element) => {
  await waitFor(() => {
    const submitBtn = container.querySelector('button[type="submit"]');
    if (submitBtn) {
      fireEvent.click(submitBtn);
    }
  });
};

export const testRouterPushCall = (
  pushFn: ReturnType<typeof jest.fn>,
  callIdx: number,
  expectedPathname: string,
  expectedSearch = '',
) => {
  expect(pushFn.mock.calls[callIdx][0].pathname).toEqual(expectedPathname);
  expect(pushFn.mock.calls[callIdx][0].search).toEqual(expectedSearch);
};

type RenderAppPartOptions = {
  push?: ReturnType<typeof jest.fn>;
  pathname?: string;
  search?: string;
  withoutThemeProvider?: boolean;
  withoutRouter?: boolean;
  store?: MockStoreEnhanced<object, object>;
  onlyReturnWrappedElement?: boolean;
};

export const renderAppPart = (element: ReactElement, options: RenderAppPartOptions = {}) => {
  const navigator = {
    createHref: jest.fn(),
    push: options.push || jest.fn(),
    go: jest.fn(),
    replace: jest.fn(),
  };

  const location = {
    pathname: options.pathname || '/',
    search: options.search || '',
  };

  let content = element;
  if (!options.withoutThemeProvider) {
    content = <ThemeProvider theme={theme}>{element}</ThemeProvider>;
  }
  if (options.store) {
    content = <Provider store={options.store}>{content}</Provider>;
  }
  if (!options.withoutRouter) {
    content = (
      <Router location={location} navigator={navigator}>
        {content}
      </Router>
    );
  }

  return options.onlyReturnWrappedElement ? content : render(content);
};
