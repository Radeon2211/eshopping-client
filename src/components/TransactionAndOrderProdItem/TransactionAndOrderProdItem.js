import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import * as SC from './TransactionAndOrderProdItem.sc';
import theme from '../../styled/theme';
import ProductThumbnail from '../UI/ProductThumbnail/ProductThumbnail';
import PlainText from '../UI/PlainText';
import { formatPrice, roundOverallPrice } from '../../shared/utility/utility';
import FlexWrapper from '../UI/FlexWrapper';

const TransactionAndOrderProdItem = React.memo(({ data, orderId }) => {
  const { _id: productId, name, price, quantity, photo } = data;

  const roundedOverallPrice = roundOverallPrice(price * quantity);

  return (
    <SC.Wrapper data-testid="TransactionAndOrderProdItem">
      <FlexWrapper $spacing="2">
        <Link
          to={`/product/${productId}`}
          data-testid="TransactionAndOrderProdItem-product-link-photo"
        >
          <ProductThumbnail
            photo={photo}
            alt={name}
            productId={productId}
            width="5"
            height="5"
            orderId={orderId}
          />
        </Link>
        <SC.NameAndPrice>
          <span className="name">
            <Link
              to={`/product/${productId}`}
              data-testid="TransactionAndOrderProdItem-product-link-name"
            >
              {name}
            </Link>
          </span>
          <FlexWrapper $justify="space-between" $align="center" $flex="1">
            <PlainText
              $size="2"
              $color={theme.colors.light4}
              data-testid="TransactionAndOrderProdItem-product-price-per-piece"
            >
              {`${quantity} x ${formatPrice(price)}`}
            </PlainText>
            <PlainText $size="4" data-testid="TransactionAndOrderProdItem-product-overall-price">
              {formatPrice(roundedOverallPrice)}
            </PlainText>
          </FlexWrapper>
        </SC.NameAndPrice>
      </FlexWrapper>
    </SC.Wrapper>
  );
});

TransactionAndOrderProdItem.defaultProps = {
  orderId: '',
};

TransactionAndOrderProdItem.propTypes = {
  data: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    quantity: PropTypes.number.isRequired,
    photo: PropTypes.bool.isRequired,
  }).isRequired,
  orderId: PropTypes.string,
};

export default TransactionAndOrderProdItem;

TransactionAndOrderProdItem.displayName = 'TransactionAndOrderProdItem';
