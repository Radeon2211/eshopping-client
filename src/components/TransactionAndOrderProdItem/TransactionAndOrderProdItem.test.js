import React from 'react';
import { mount } from 'enzyme';
import { Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import TransactionAndOrderProdItem from './TransactionAndOrderProdItem';
import ProductThumbnail from '../UI/ProductThumbnail/ProductThumbnail';
import * as SC from './TransactionAndOrderProdItem.sc';
import theme from '../../styled/theme';
import { checkProps, createTransactionAndOrderProdItem } from '../../shared/testUtility';
import { baseURL } from '../../axios';

const defaultHistory = {
  listen: jest.fn(),
  createHref: jest.fn(),
  location: { pathname: '/cart' },
};

const setUp = (data, orderId = '') => {
  const props = {
    data,
    orderId,
  };
  return mount(
    <Router history={defaultHistory}>
      <ThemeProvider theme={theme}>
        <TransactionAndOrderProdItem {...props} />
      </ThemeProvider>
    </Router>,
  );
};

describe('<TransactionAndOrderProdItem />', () => {
  describe('Check prop types', () => {
    it('Should NOT throw a warning', () => {
      const props = {
        data: { _id: 'i1' },
      };
      expect(checkProps(TransactionAndOrderProdItem, props)).toBeUndefined();
    });

    it('Should throw a warning', () => {
      expect(checkProps(TransactionAndOrderProdItem, {})).not.toBe(null);
    });
  });

  describe('Checks how everything render', () => {
    it('Should render correctly with full data', () => {
      const data = createTransactionAndOrderProdItem(
        'u1',
        'user1',
        'p1',
        2,
        9.9,
        'productName',
        true,
      );
      const wrapper = setUp(data);
      expect(wrapper.find(SC.Wrapper)).toHaveLength(1);
      expect(wrapper.find('[data-test="product-link"]').length).toBeGreaterThan(1);
      expect(
        wrapper.find('[data-test="product-link"]').every((link) => {
          return link.prop('to') === '/product/p1';
        }),
      );
      expect(wrapper.find(ProductThumbnail)).toHaveLength(1);
      expect(wrapper.find(SC.NameAndPrice)).toHaveLength(1);
      expect(wrapper.find('.name')).toHaveLength(1);
      expect(wrapper.find('.name').text()).toEqual('productName');
      expect(wrapper.find(SC.Price)).toHaveLength(1);
      expect(wrapper.find('.price-per-piece').length).toBeGreaterThan(1);
      expect(wrapper.find('.price-per-piece').first().text()).toEqual('2 x $9.90');
      expect(wrapper.find('.overall-price')).toHaveLength(1);
      expect(wrapper.find('.overall-price').text()).toEqual('$19.80');
    });

    it('Should render 1 x price per piece', () => {
      const data = createTransactionAndOrderProdItem(
        'u1',
        'user1',
        'p1',
        1,
        9.9,
        'productName',
        false,
      );
      const wrapper = setUp(data);
      expect(wrapper.find('.price-per-piece').first().text()).toEqual('1 x $9.90');
      expect(wrapper.find('.overall-price').text()).toEqual('$9.90');
    });
  });
});
