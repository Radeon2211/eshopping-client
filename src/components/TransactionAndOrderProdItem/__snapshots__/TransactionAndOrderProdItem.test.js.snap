// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionAndOrderProdItem /> Checks how renders and behaviour should render everything correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ ZqsAz sc-gsTCUz inazTm"
    data-testid="TransactionAndOrderProdItem"
    direction="row"
    spacing="2"
    width="100%"
    wrap="nowrap"
  >
    <a
      data-testid="TransactionAndOrderProdItem-product-link-photo"
    >
      <div
        class="sc-bdfBwQ igmTYO sc-eCssSg eawKUo"
        data-testid="ProductThumbnail"
        direction="row"
        height="5"
        spacing=""
        width="5"
        wrap="nowrap"
      >
        <img
          alt="productName"
          data-testid="ProductThumbnail-img"
          src="http://192.168.1.109:4000/products/p1/photo"
        />
      </div>
    </a>
    <div
      class="sc-dlfnbm cjrVjg"
    >
      <span
        class="name"
      >
        <a
          data-testid="TransactionAndOrderProdItem-product-link-name"
        >
          productName
        </a>
      </span>
      <div
        class="sc-bdfBwQ lgjdug sc-hKgILt huHHva"
        direction="row"
        spacing=""
        width="100%"
        wrap="nowrap"
      >
        <span
          class="sc-jSgupP jUhfHv"
          color="#777"
          display="inline-block"
          overflow="visible"
          size="2"
          spacing="0px"
          transform=""
        >
          2 x $9.90
        </span>
        <span
          class="sc-jSgupP jiAuFg"
          color=""
          display="inline-block"
          overflow="visible"
          size="4"
          spacing="0px"
          transform=""
        >
          $19.80
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
